### ISSUE TITLE
Local search: review html markup, should be semantic

### ISSUE THREAD

**Issue by userquin** [vuejs/vitepress#2232]
### Describe the bug

The search button and dialog should use semantic markup: roles and aria attributes

### Reproduction

https://unocss.dev/

### Expected behavior

Just check Algolia search button and dialog on VP docs.

### System Info

```shell
NA
```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
